packages/esm-patient-orders-app/src/lab-results/lab-results-form.test.tsx

```diff
@@ -0,0 +1,258 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { useOrderConceptByUuid, useLabEncounter, useObservation } from './lab-results.resource';
+import LabResultsForm from './lab-results-form.component';
+import { type Order } from '@openmrs/esm-patient-common-lib';
+
+jest.mock('./lab-results.resource', () => ({
+  useOrderConceptByUuid: jest.fn(),
+  useLabEncounter: jest
+    .fn()
+    .mockReturnValue({ encounter: [{ obs: [] }], isLoading: false, mutateLabOrders: jest.fn() }),
+  useObservation: jest.fn().mockReturnValue({ data: [], isLoading: false }),
+  updateOrderResult: jest.fn().mockReturnValue(Promise.resolve()),
+}));
+
+jest.mock('@openmrs/esm-framework', () => ({
+  useAbortController: jest.fn(() => ({ signal: new AbortController().signal })),
+  useLayoutType: jest.fn(() => 'desktop'),
+  showSnackbar: jest.fn(),
+}));
```

Comment:
```suggestion
```

This can be removed and in their stead we can rely on the default stubs from the framework. See https://o3-docs.openmrs.org/docs/frontend-modules/unit-and-integration-testing#mocking-patterns.